test/tests/functional/pbs_release_limited_res_suspend.py

```diff
@@ -0,0 +1,352 @@
+# coding: utf-8
+
+# Copyright (C) 1994-2016 Altair Engineering, Inc.
+# For more information, contact Altair at www.altair.com.
+#
+# This file is part of the PBS Professional ("PBS Pro") software.
+#
+# Open Source License Information:
+#
+# PBS Pro is free software. You can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# The PBS Pro software is licensed under the terms of the GNU Affero General
+# Public License agreement ("AGPL"), except where a separate commercial license
+# agreement for PBS Pro version 14 or later has been executed in writing with
+# Altair.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software - under
+# a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+import time
+from tests.functional import *
+
+
+class TestReleaseLimitedResOnSuspend(TestFunctional):
+    """
+    Test that based on admin's input only limited number of resources are
+    released from a running job.
+    """
+
+    def setUp(self):
+        TestFunctional.setUp(self)
+        # Set default resources available on the default mom
+        a = {'resources_available.ncpus': 4, 'resources_available.mem': '2gb'}
+        self.server.manager(MGR_CMD_SET, NODE, a, self.mom.shortname)
+        # Create an express queue
+        b = {'queue_type': 'Execution', 'enabled': 'True',
+             'started': 'True', 'priority': '200'}
+        self.server.manager(MGR_CMD_CREATE, QUEUE, b, "expressq")
+
+    def test_do_not_release_mem_sched_susp(self):
+        """
+        During preemption by suspension test that only ncpus are released from
+        the running job and memory is not released.
+        """
+
+        # Set restrict_res_to_release_on_suspend server attribute
```

Comment:
I'm not sure I find this comment useful.  The code itself says that you're setting the attribute.  It's not really harmful though.